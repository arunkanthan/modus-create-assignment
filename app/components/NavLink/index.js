// @flow
import * as React from 'react';
import { NavLink } from 'react-router-dom';

type NavLinkProps = {
  to: string,
  label: string,
  styles: Object,
};

const StyledNavLink = ({ to, label, styles }: NavLinkProps) => (
  <NavLink to={to} className={styles.navLink} activeClassName={styles.selected}>
    {label}
  </NavLink>
);

export default StyledNavLink;
