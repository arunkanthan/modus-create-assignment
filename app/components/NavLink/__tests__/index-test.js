import React from 'react';
import renderer from 'react-test-renderer';
import { HashRouter as Router } from 'react-router-dom';
import NavLink from '..';

it('renders correctly', () => {
  const tree = renderer
    .create(
      <Router>
        <NavLink to="test" label="test" styles={{}} />
      </Router>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
