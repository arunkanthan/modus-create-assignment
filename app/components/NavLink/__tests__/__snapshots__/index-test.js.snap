// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<a
  aria-current={null}
  href="#/test"
  onClick={[Function]}
>
  test
</a>
`;
